src/middlewared/middlewared/plugins/zettarepl.py

```diff
@@ -290,9 +292,11 @@ def stop(self):
         with self.lock:
             if self.process:
                 self.process.terminate()
-                self.process.join(5)
-
-                if self.process.is_alive():
+                for i in range(50):
```

Comment:
Does linux have a facility to wait for a process? Potentially sending 50 OS signals and another 50 os sleep syscalls could be avoided